File: 121. Best Time to Buy and Sell Stock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let lowestPrice = prices[0];
  let bestValue = 0;
  prices.forEach(price => {
    lowestPrice = Math.min(price, lowestPrice);
    bestValue = Math.max(price - lowestPrice, bestValue);
  });

  return bestValue;
};

File: 121. Best Time to Buy and Sell Stock/solution.go
func maxProfit(prices []int) int {
	min, profit := 0, 0
	for i, price := range prices {
		if i == 0 {
			min = price
			continue
		}

		if diff := price - min; diff > profit {
			profit = diff
		}
		if price < min {
			min = price
		}
	}
	return profit
}